src/lib/model.ts

```diff
@@ -82,6 +83,8 @@ export function fromObservable(target: Model, propertyKey: string): void {
 
         this[subPropertyKey].add(observableForProperty.subscribe(
           (x) => {
+            if (isEqual(this[valPropertyKey], x)) return;
```

Comment:
So basically in order to make most of this work, we've fallen back to doing deep comparisons for everything. It's not ideal, but I think in the usual case it'll be okay